### ISSUE TITLE
Add API for returning current network connected by wallet

### ISSUE THREAD

**Issue by 314159265359879** [blockstack/blockstack-app#4194]
For mainnet, public testnet, subnets (formerly hyperchains) and custom testnets. And to avoid issues of the past where users forget they were connected to testnet with the wallet and are unable to sign transactions because it a different network then the website/dapp. 

A. Dapp should be able to get current network connected to by the wallet
B. Dapp should be able to request for an addition https://github.com/leather-wallet/extension/issues/2535
C. Dapp should be able to send request for the wallet to switch to a different network entirely. https://github.com/leather-wallet/extension/issues/4193

**Comment by markmhendrickson** [comment]
Thanks @314159265359879, I've simplified the title to focus on this issue's particular contribution to making the app-related network management UX better.

**Comment by 314159265359879** [comment]
tracked via https://github.com/leather-io/issues/issues/279

**Comment by ufe-pr** [comment]
Some users of [LaserEyes](https://github.com/omnisat/lasereyes-mono) are also experiencing issues as a result of this. Our connection logic depends on detecting the current network, and the lack of a programmatic way to retrieve it is leading to integration problems across environments. We'd really appreciate any updates or potential workarounds the team can provide here

**Comment by markmhendrickson** [comment]
@ufe-pr is there any particular interface standard you'd like us to apply here?

**Comment by kyranjamie** [comment]
@ufe-pr With our new RPC APIs, we default to mainnet, and you'll have to pass the network type if you wish to connect to another. As such, if the app itself has requested and thus knows which is connected, is this still an issue?

Any more details about your use case would be super helpful

**Comment by ufe-pr** [comment]
@markmhendrickson , we don't have any particular preference. Majorly, we'll just like to be able to determine the user's currently active network

**Comment by ufe-pr** [comment]
@kyranjamie while this is better, it helps to still be able to request the current network. This is because we typically persist the connection state in local storage to avoid repeated pop-ups on page reload. This state can easily get stale as the user might switch the network between page visits.

If the wallet wouldn't always pop-up for an already connected site (when requesting addresses), then it's perfectly fine

**Comment by kyranjamie** [comment]
> @ufe-pr This state can easily get stale as the user might switch the network between page visits.

But, if Leather persists that requests from `www.lasereyes.build` came from account `3` on `testnet`, and we route future requests to these, then it doesn't matter if the user changes network/account, right?

A case where we do need a network endpoint, is selecting a specific network, not just the mode. I might have 3 devnets, for instance. My thinking on this is that on `getAddresses` you choose a network _mode_, then we have a `getNetworks` that shows all networks of the selected mode (intentionally omitting other networks, if a user selects a devnet, they've no business  seeing what mainnets are configured).

**Comment by ufe-pr** [comment]
> But, if Leather persists that requests from `www.lasereyes.build` came from account `3` on `testnet`, and we route future requests to these, then it doesn't matter if the user changes network/account, right?

I agree with this. I think if leather persists that information, then it doesn't matter
